apollo-router/src/plugins/telemetry/mod.rs

```diff
@@ -1245,6 +1251,58 @@ impl Telemetry {
         };
         sender.send(SingleReport::Stats(metrics));
     }
+
+    fn decode_subgraph_ftv1(context: &Context) -> (u64, HashMap<String, SingleTypeStat>) {
+        use crate::plugins::telemetry::apollo_exporter::proto::reports::trace::Node;
+
+        fn traverse(
+            per_type: &mut HashMap<String, SingleTypeStat>,
+            field_execution_weight: f64,
+            node: &Node,
+        ) {
+            let single_field_stat = per_type
+                .entry(node.parent_type.clone())
+                .or_default()
+                .per_field_stat
+                .entry(node.original_field_name.clone())
+                .or_insert_with(|| metrics::apollo::studio::SingleFieldStat {
+                    return_type: node.r#type.clone(), // not `Default::default()`’s empty string
+                    errors_count: 0,
+                    latency: Default::default(),
+                    estimated_execution_count: 0.0,
+                    requests_with_errors_count: 0,
+                });
+            single_field_stat.errors_count += node.error.len() as u64;
+            single_field_stat.latency.increment_duration(
+                Duration::from_nanos(node.end_time.saturating_sub(node.start_time)),
+                field_execution_weight,
+            );
+            single_field_stat.estimated_execution_count += field_execution_weight;
```

Comment:
update `observed_execution_count` too?

Reply:
This struct doesn’t have that field, instead conversion to protobuf types fills it with the total count from the duration historgram:

https://github.com/apollographql/router/blob/ff14c473bf/apollo-router/src/plugins/telemetry/metrics/apollo/studio.rs#L285

Reply:
@SimonSapin I don't think that works because that `total` is also incremented here by `field_execution_weight`. In fact I think that `field_stat.estimated_execution_count` always ends up equal to `field_stat.latency.total` here. I think you should be tracking *observed* count (ie, the literal number of times you see that this happened, not weighted) separately and calculating *estimated* count based on `latency.total`... either that or change `increment_bucket` to increment total by 1 rather than `value` (though that might affect other callers too).

Reply:
Sorry I missed that comment! For anyone else following along, this was fixed in https://github.com/apollographql/router/pull/2598